test/test_jit.py

```diff
@@ -1509,17 +1555,19 @@ def func(a, b):
 
     def test_script_for_in_range(self):
         def fn():
-            c = 0
+            c = FIXME_zerol()
             for i in range(100):
                 c += i
             return c
         self.checkScript(fn, (), outputs=4950, optimize=True)
 
     def test_script_for_in_range_dynamic(self):
         def fn():
-            c = 0
+            c = FIXME_zerol()
```

Comment:
Is the `0` literal so broken right now that we can't even use it? Why?

Reply:
Yes, the 0 literal is broken for a number of reasons. In this particular case the problem is that `for i in range(100)` assigns `i` a TensorType. In the above test (test_script_for_in_range), we're using `c` as the accumulator and adding `i` to `c`.

If we assign `c = 0` originally, `c` has type IntType. However, `i` has some tensor type, and we disallow reassigning variables to things that have unrelated types. This can be fixed when the `range(...)` expression gives an IntType.

I can add more comments to the code if you think that is fine